src/formatDate.js

```diff
@@ -50,7 +50,76 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const currentSep = fromFormat[3];
+  const finalSep = toFormat[3];
+  const dateBefore = date.split(currentSep);
+  const dateAfter = [];
+
+  let day;
+  let month;
+  let year;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = dateBefore[i];
+        break;
+
+      case 'MM':
+        month = dateBefore[i];
+        break;
+
+      case 'YY':
+        year = dateBefore[i];
+        break;
+
+      case 'YYYY':
+        year = dateBefore[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (const newFormat of toFormat) {
+    switch (newFormat) {
+      case 'DD':
+        dateAfter.push(day);
+        break;
+
+      case 'MM':
+        dateAfter.push(month);
+        break;
+
+      case 'YY':
+        dateAfter.push(yearFormat(year, newFormat));
+        break;
+
+      case 'YYYY':
+        dateAfter.push(yearFormat(year, newFormat));
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return dateAfter.join(finalSep);
+}
+
+function yearFormat(year, format) {
```

Comment:
any ideas on improving the name of this function?

Reply:
I think formatYear or formatTheYear is better because it is a function. I will improve it.

Reply:
Well done)